test/Microsoft.IdentityModel.TestUtils/IdentityComparer.cs

```diff
@@ -1669,5 +1669,14 @@ private static string GetString(object str)
             else
                 return string.Format(CultureInfo.InvariantCulture, "{0}", (str ?? "null"));
         }
+
+        public static bool AreDatesEqualWithEpsilon(DateTime? dateTime1, DateTime? dateTime2, int epsilon)
```

Comment:
Shouldn't `epsilon` come from the `CompareContext` instead? Then pass the `CompareContext` here.

Reply:
Why pass nullable?